dm/pkg/checker/conn_checker.go

```diff
@@ -114,7 +131,13 @@ func (c *connNumberChecker) check(ctx context.Context, checkerName string) *Resu
 		result.State = StateFailure
 	} else {
 		// nonzero max_connections and needed connections are less than or equal to max_connections
-		result.State = StateSuccess
+		if result.State != StateWarning {
+			// if no enough privilege to check the user's connection number,
+			// the result state will be StateWarning and we can't mark it as success.
+			result.State = StateSuccess
```

Comment:
it is better to assign `StateSuccess` in L51

Reply:
962d7de

Reply:
And then remove L134-138?

Reply:
c79c0e0